packages/rainbowkit/src/components/ConnectOptions/ConnectDetails.tsx

```diff
@@ -231,21 +229,24 @@ export function ConnectDetail({
     window.open(uri, safari ? '_blank' : '_self');
   };
 
+  const isWalletConnect = wallet.id === 'walletConnect';
+
   const secondaryAction: {
     description: string;
     label: string;
     onClick?: () => void;
     href?: string;
-  } | null = showWalletConnectModal
+    icon?: JSX.Element;
+  } | null = isWalletConnect
     ? {
         description: !compactModeEnabled
           ? i18n.t('connect.walletconnect.description.full')
           : i18n.t('connect.walletconnect.description.compact'),
-        label: i18n.t('connect.walletconnect.open.label'),
+        label: i18n.t('connect.walletconnect.copy.label'),
         onClick: () => {
-          onClose();
-          showWalletConnectModal();
+          if (qrCodeUri) navigator.clipboard.writeText(qrCodeUri);
```

Comment:
Wrap navigator.clipboard in optionals; do this for the ProfileDetails copy implementation as well